### ISSUE TITLE
Accessing 'undefined' value in passStringToWasm -- Chrome autocomplete

### ISSUE THREAD

**Issue by martin-kolarik** [rustwasm/wasm-bindgen#2746]
When Chrome puts autocomplete value into <input "text">, it creates keydown event with key property undefined. Using KeyboardEvent::key() then asks for string value, passStringToWasm is called, and there arg 'undefined'.length is accessed.

Empty string should be returned to caller in this case.

**Comment by jonboj** [comment]
The source code refered above, is this one of the `wasm-bindgen/examples`, in case which one?

**Comment by martin-kolarik** [comment]
Not know, I run my project. I am using dominator's (https://github.com/Pauan/rust-dominator) event handling, where mentioned call appears> https://github.com/Pauan/rust-dominator/blob/1f541fd7170287e657c4cae1ff418430cfd625ed/src/events.rs#L131.

Dominator itself is only accessing key() method of web-sys::KeboardEvent, then the issue happens in code generated by wasm-bindgen (AFAIK).

**Comment by jonboj** [comment]
Looking at [SO autocomplete Chrome](https://stackoverflow.com/questions/59534586/google-chrome-fires-keydown-event-when-form-autocomplete), running the `JSFiddle` snippet in the question, when filling with autocomplete `console.log (e.key);` logs `undefined` in the JSFiddle console. So to me it would be unexpected if `web-sys::KeboardEvent.key()` didn't return same.

**Comment by martin-kolarik** [comment]
Yes, I confirm that running the JSFiddle reports correctly 'e.key = undefined'. I think this is the point. According to https://www.w3.org/TR/uievents/#idl-keyboardevent, 'key' property must always be defined, but Chrome, when autocompleting, constructs fake keydown event without the property defined:
![image](https://user-images.githubusercontent.com/74831268/147971673-84f74182-7434-4477-aaff-42dfe5f83888.png)

I recognize that wasm-bindgen's IDLs are correct, and that Chrome behaves wrongly (IMO). Chrome injects bad event, wasm-bindgen's generated key accessor processes the event, expecting it cannot be undefined, and as a result wasm-bindgen's passStringToWasm fails.

Formally incorrect (but working) solution would be to change IDL from `readonly attribute DOMString key;` to `readonly attribute DOMString? key;`. Then gen_KeyboardEvent.rs would contain `key(...) -> Option<String>` and JS interop would not fail. I have tested this approach and it works.

So, now it looks to me like that wasm-bindgen does not manage to defend against malformed Chrome event.

**Comment by jonboj** [comment]
Great it is consistent! Not a developer of `wasm-bindgen` - So an opinionated observation. Maybe publish a minimal repo reproducing the problem with `passStringToWasm`.

 Actually dominator has a TODO at line above [dominator TODO](https://github.com/Pauan/rust-dominator/blob/1f541fd7170287e657c4cae1ff418430cfd625ed/src/events.rs#L130),  which mention `enum` instead of `String`, so could indicate implementor was aware of problems with using a `String` value.
